### PR TITLE
correct timebase in sys_arch.c

### PR DESCRIPTION
**Pull request by RLuckom**
sys_now() now returns ms.

### CODE REVIEW

File: lwip/sys_arch.c [hunk 1]
```diff
@@ -210,7 +210,7 @@ unsigned long ulReturn;
 	{
 		configASSERT( xInsideISR == ( portBASE_TYPE ) 0 );
 
-		if( pdTRUE == xQueueReceive( *pxMailBox, &( *ppvBuffer ), ulTimeOut/ portTICK_RATE_MS ) )
+		if( pdTRUE == xQueueReceive( *pxMailBox, &( *ppvBuffer ), ulTimeOut ) )
```
Comment (RLuckom) [inline lwip/sys_arch.c]: ulTimeout is supposed to be in ms. If so, it should not be divided by portTICK_RATE_MS

**Comment by RLuckom** [comment]
See https://github.com/SuperHouse/esp-open-rtos/issues/123 for the long-winded rationale.

File: lwip/sys_arch.c (hunk 1 above)
Comment (RLuckom) [inline lwip/sys_arch.c]: Sorry, I think I misunderstood portTICK_RATE_MS. This was correct before.

**Comment by projectgus** [comment]
Welcome to the project, thanks for looking into this so thoroughly!

Looks good! And you don't think we've accidentally depended on the existing behaviour in any other places?

The one place I know about is here:
https://github.com/SuperHouse/esp-open-rtos/blob/master/extras/paho_mqtt_c/MQTTESP8266.c#L73

... but that might be the only one?

**Comment by RLuckom** [comment]
Thanks for pointing that out.

I haven't actually found a reason for the select() bug referenced in issue 134. As far as I can tell, select calls sys_arch_sem_wait (which handles ms correctly), and sys_arch_sem_wait converts ms to ticks and then calls into xSemaphoreTake, which is defined to be an invocation of xQueueGenericReceive (and is supposed to take a timeout in ticks).

Because the mqtt extra is calling select, and I haven't found a connection between this change and select, I'm not sure I should add that change. However I'll keep looking into it.

**Comment by projectgus** [comment]
Thanks for the detailed investigation and explanation @RLuckom . Sorry I didn't get back to you earlier.

As I commented on the parent issue, I agree with you that there isn't actually a bug with select(). However I think this is also a legitimate bug fix, so I've cherry-picked it into master as 0734fa4166.

Thanks again!

**Pull request closed without merging**